public/gatk-engine/src/main/java/org/broadinstitute/gatk/engine/datasources/reads/GATKBAMIndex.java

```diff
@@ -47,116 +34,24 @@
  * @author mhanna
  * @version 0.1
  */
-public class GATKBAMIndex {
+public abstract class GATKBAMIndex {
+
     /**
-     * BAM index file magic number.
+     * What is the starting bin for each level?
      */
-    private static final byte[] BAM_INDEX_MAGIC = "BAI\1".getBytes();
+    protected static final int[] LEVEL_STARTS = {0,1,9,73,585,4681};
```

Comment:
`LEVEL_STARTS` static array was in original `GATKBAMIndex`. The only thing that was changed here is: private -> protected
I think for an explanation we better ask its original author `mhanna`